### ISSUE TITLE
Entire cluster incorrectly labeled as noise for specific random seeds

### ISSUE THREAD

**Issue by SamuelO64** [TutteInstitute/fast_hdbscan#63]
### Description

Thank you for your work on `fast_hdbscan`. I am currently using it to separate point clouds into distinct objects that all move toward a common center.

As the objects move over time, I occasionally observe that an entire cluster is classified as noise on certain frames, with no apparent reason, and only for specific parameter values (e.g. `min_cluster_size` or `cluster_selection_persistence`) — even when those values should not be breaking the clusters.

### Reproduction

In my code, Gaussian point clouds are placed randomly using `rng = np.random.default_rng(seed)`. Sweeping `seed` from 0 to 1000, I only encountered the bug for:
- `seed = 700`
- `seed = 727`

The same parameters **do not** trigger the issue with `sklearn.cluster.HDBSCAN`.

### Parameters used

```python
min_cluster_size=192,
min_samples=1,
cluster_selection_epsilon=0.05957,
allow_single_cluster=True
```

### Additional context

I also implemented a temporal tracking method to maintain consistent labels across frames. It works by clustering both the current frame and the previous frame together simultaneously. This part appears to work correctly with `sklearn`'s HDBSCAN.

I am attaching the reproduction code below. The helpers are:
- `generate_all_frames` — generates the synthetic data
- `visualize_pcd_temporal` — creates the Open3D visualization, coloring noise points in black

[visu_tempo_suivi_gauss_eng.py](https://github.com/user-attachments/files/26628447/visu_tempo_suivi_gauss_eng.py)

I am also attaching screenshots showing the bug for seed 700 and seed 727

Seed 700 - frame 29
<img width="912" height="830" alt="Image" src="https://github.com/user-attachments/assets/6f62836f-b50a-491b-85dc-70d7a375a457" />
frame 30
<img width="841" height="811" alt="Image" src="https://github.com/user-attachments/assets/94505bf9-a5fb-47a1-bec3-ce7b2368cd12" />
frame 31
<img width="885" height="814" alt="Image" src="https://github.com/user-attachments/assets/7cdf7640-8db5-4b35-802d-84ab083dfdc4" />

Seed 727 - frame 30
<img width="851" height="842" alt="Image" src="https://github.com/user-attachments/assets/ecac2cdc-5f30-448b-a99c-23b17acdb332" />
frame 31
<img width="833" height="822" alt="Image" src="https://github.com/user-attachments/assets/ad70a2e5-7485-4a1b-a82d-d41a7be05bf9" />
frame 32
<img width="769" height="811" alt="Image" src="https://github.com/user-attachments/assets/fe386f64-d062-419a-aa7f-978d33bd996b" />

### Expected behavior

The cluster should be identified correctly, consistent with the behavior of `sklearn.cluster.HDBSCAN` under the same parameters.

### Actual behavior

The entire cluster is labeled as noise (`-1`) for certain seeds, even though the same parameter values work correctly in most cases and with `sklearn`.

### Environment

- `fast_hdbscan` version: 0.3.2 (same bug for all versions)
- `sklearn` version: 1.7.2
- Python version: 3.10.11
- OS: Windows 10

Thank you !
